dist/logs/logzioLogsFunction/core.js

```diff
@@ -1,60 +1,186 @@
-  metrics: {
-    token: process.env.LogzioMetricsToken,
-    host: process.env.LogzioMetricsHost,
-  },
-  logs: {
-    token: process.env.LogzioLogsToken,
-    host: process.env.LogzioLogsHost,
-  },
-});
-
-module.exports = function processEventHubMessages(context, eventHubMessages, {
-  enableMetric = false,
-}) {
-  const {
-    host,
-    token,
-  } = enableMetric ? getParserOptions().metrics : getParserOptions().logs;
-  context.log(`Starting Logz.io Azure function with enableMetrics: ${enableMetric}`);
-  context.log(JSON.stringify(eventHubMessages));
-  const callBackFunction = getCallBackFunction(context);
-
-  const logzioShipper = logger.createLogger({
-    token,
-    host,
-    type: 'eventHub',
-    protocol: 'https',
-    internalLogger: context,
-    compress: true,
-    debug: true,
-    callback: callBackFunction,
-  });
-
-  const dataParser = new DataParser({
-    internalLogger: context,
-    enableMetric,
-  });
+const uploadFiles = async (containerClient, filesToUpload, context) => {
+    try {
+        const promises = [];
+        for (const file of filesToUpload) {
+            const blockBlobClient = containerClient.getBlockBlobClient(file);
+            promises.push(blockBlobClient.uploadFile(file));
+        }
+        await Promise.all(promises);
+        if (!arrayIsEmpty(filesToUpload)) {
+            context.log('Uploaded logs to back up container.');
+        }
+    } catch (error) {
+        context.log('Error: ', error);
+    }
+};
+const writeEventToBlob = async (event, filesToUpload, foldersToDelete, context) => {
+    if (logIndex >= maxBatchSize) {
+        fileName = createFileName();
+        updateFolderSize();
+        logIndex = 1;
+    } else {
+        logIndex++;
+        const eventWithNewLine = JSON.stringify(event).concat('\n');
+        const concatFolderToFile = folderName + "\\" + fileName;
+        const filePath = workingDir + "\\" + concatFolderToFile;
+        try {
+            await appendFileAsync(filePath, eventWithNewLine);
+            pushFileToUploadAsync(filesToUpload, concatFolderToFile, context);
+            pushFolderToDeleteAsync(foldersToDelete, folderName, context);
+        }
+        catch (e) {
+            context.log(`Error was thrown in appendFile, ${e}`);
+        }
+    };
+};
+const pushFileToUploadAsync = (filesToUpload, filePath, context) => {
+    if (!filesToUpload.includes(filePath)) {
+        filesToUpload.push(filePath);
+    }
+}
+const pushFolderToDeleteAsync = (foldersToDelete, folderPath, context) => {
+    if (!foldersToDelete.includes(folderPath)) {
+        foldersToDelete.push(folderPath);
+    }
+}
 
-  const parseMessagesArray = dataParser.parseEventHubLogMessagesToArray(eventHubMessages);
-  context.log(`About to send ${parseMessagesArray.length} logs...`);
-  parseMessagesArray.forEach((log) => {
-    context.log(JSON.stringify(`logging: ${log}`));
-    logzioShipper.log(log);
-  });
+const updateFolderNameIfFull = () => {
```

Comment:
```suggestion
const updateFolderNameIfMaxSizeSurpassed = (folderMaxSizeInMB = 1000) => {
    if (folderSize >= folderMaxSizeInMB) {
        folderName = createFolderName();
        folderSize = 0;
    }
}
```
and remove MB declaration